opennms-webapp/src/main/java/org/opennms/web/tags/DateTimeTag.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.tags;
+
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static java.time.temporal.ChronoField.HOUR_OF_DAY;
+import static java.time.temporal.ChronoField.MINUTE_OF_HOUR;
+import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.servlet.jsp.tagext.SimpleTagSupport;
+
+/**
+ * This class replaces the &lt;fmt:formatDate /&gt; tag.
+ * Why do we need a new tag?
+ * => fmt can't be configured via a System Property (without side effects)
+ * => we want to support the new java.time classes
+ *
+ * It will output datetimes as ISO_8601 type style unless otherwise defined in opnnms.properties.
+ * See also:
+ *   https://en.wikipedia.org/wiki/ISO_8601 and
+ *   https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html
+ */
+public class DateTimeTag extends SimpleTagSupport {
+
+    final static String SYSTEM_PROPERTY_DATE_FORMAT = "org.opennms.ui.datettimeformat";
+    private final static Logger log = Logger.getLogger(DateTimeTag.class.getName());
```

Comment:
We have the convention that static variables are starting with a upper case letter.
If it is also final , it is considered a constant and should be all upper case: `LOG` vs `log` in this case.

Reply:
ok I will change that. I now also see that the other Loggers are upper case.
Btw. there are different schools of thought on that topic, see: https://stackoverflow.com/questions/1417190/should-a-static-final-logger-be-declared-in-upper-case but I guess the most important one is to be consistent.